test/e2e/infra/secondary_network/secondary_network_prerequisite.sh

```diff
@@ -0,0 +1,361 @@
+}
+#This function is not required to be called if the --interface-name and --sriov-vf-count params are not provided.
+function configureVFs() {
+	checkInterface
+	if [[ $(cat /sys/class/net/$interface/device/sriov_numvfs) != 0 ]]; then
+		echo "0" >/sys/class/net/$interface/device/sriov_numvfs
+		sleep 2
+	fi
+	if [[ $configure -eq true ]]; then
+		echo "$vfs_num" >/sys/class/net/$interface/device/sriov_numvfs
+		cat /sys/class/net/$interface/device/sriov_numvfs
+	else
+		echo "0" >/sys/class/net/$interface/device/sriov_numvfs
+		cat /sys/class/net/$interface/device/sriov_numvfs
+	fi
+}
+
+#This function is not required to be called if the --interface-name and --sriov-vf-count params were not provided.
+function setIPLinkUp() {
+	checkInterface
+	ip link set $interface up
+	if [ $? -ne 0 ]; then
+		echoerr "Failed to set Ip link up!"
+		exit 1
+	fi
+	ifconfig $interface
+}
+
+function downloadSecondaryNetworkPrerequisiteYAMLFiles() {
+	rm -rf $SEC_NET_LOCAL_CONFIG_DIR_PATH
+	mkdir -p $SEC_NET_LOCAL_CONFIG_DIR_PATH
+	cp $THIS_DIR/$YAML_NET_ATTACH_DEF_CRD $SEC_NET_LOCAL_CONFIG_DIR_PATH
+	cp $THIS_DIR/$YAML_VIRTUAL_NET_INSTANCE $SEC_NET_LOCAL_CONFIG_DIR_PATH
+	if [ -z $interface ] && [ -z $vfs_num ]; then
+		urls=("$URL_WHEREABOUTS_IP_POOLS" "$URL_WHEREABOUTS_OVERLAPPING_IP_RANGE_RES")
+	else
+		urls=("$URL_ANTREA_CONFIG_MAP" "$URL_SRIOV_DP_DAEMONSET" "$URL_WHEREABOUTS_IP_POOLS" "$URL_WHEREABOUTS_OVERLAPPING_IP_RANGE_RES")
+	fi
+	for url in ${urls[@]}; do
+		wget -P $SEC_NET_LOCAL_CONFIG_DIR_PATH "$url"
+		if [ $? -ne 0 ]; then
+			echoerr "Failed to download Secondary Network Prerequisite YAML Files!"
+			exit 1
+		fi
+	done
+}
+
+#This function is not required to be called if the --interface-name and --sriov-vf-count params were not provided.
+function SriovDevicePlugin() {
+	kubectl $action -f $SEC_NET_REMOTE_CONFIG_DIR_PATH$YAML_CONFIG_MAP
+	if [ $? -ne 0 ]; then
+		echoerr "Failed to $action Sriov Device Plugin- configMap.yaml"
+		exit 1
+	fi
+
+	kubectl $action -f $SEC_NET_REMOTE_CONFIG_DIR_PATH$YAML_SRIOV_DP_DAEMONSET
+	if [ $? -ne 0 ]; then
+		echoerr "Failed $action Sriov Device Plugin - sriovdp-daemonset.yaml"
+		exit 1
+	fi
+	echo "$action ConfigMap and SriovDevicePlugin done!"
+}
+
+function NetworkAttachmentDefinition() {
+	kubectl $action -f $SEC_NET_REMOTE_CONFIG_DIR_PATH$YAML_NET_ATTACH_DEF_CRD
+	if [ $? -ne 0 ]; then
+		echoerr "Failed to $action Network Attachment Definition CRD- network-attachment-definition-crd.yaml"
+		exit 1
+	fi
+	echo "$action NetworkAttachmentDefinition Done!"
+}
+
+function VirtualNetworks() {
+	kubectl $action -f $SEC_NET_REMOTE_CONFIG_DIR_PATH$YAML_VIRTUAL_NET_INSTANCE -n $NAMESPACE
+	if [ $? -ne 0 ]; then
+		echoerr "Failed to $action Virtual Network Instance: $instance"
+		exit 1
+	fi
+	echo " $action VirtualNetworks Done!"
+}
+
+function WhereaboutsCNI() {
+	kubectl $action -f $SEC_NET_REMOTE_CONFIG_DIR_PATH$YAML_WHEREABOUTS_IP_POOL
+	if [ $? -ne 0 ]; then
+		echo "Failed to $action Whereabouts CNI"
```

Comment:
`echo` vs `echoerr`

here and in other places still

Reply:
sorry about that. I took care of it in all the err cases now, while left echo statements for the info logs.